tests/Unleash.Tests/Communication/UnleashApiClient_Features_Backoff_Tests.cs

```diff
@@ -0,0 +1,263 @@
+        httpClient.BaseAddress = new Uri("http://localhost:8080/");
+
+        var apiClient = new UnleashApiClient(
+            httpClient,
+            A.Fake<IJsonSerializer>(),
+            A.Fake<UnleashApiClientRequestHeaders>(),
+            new EventCallbackConfig()
+        );
+
+        // Act
+        var result = Task.Run(() => apiClient.FetchToggles("etag", CancellationToken.None))
+            .GetAwaiter()
+            .GetResult();
+        
+        // Assert
+        messageHandler.CallCount.Should().Be(1);
+    }
+
+    [Test]
+    public void NotModified_Should_Not_Skip_Next()
+    {
+        // Arrange
+        var messageHandler = new CountingConfigurableHttpMessageHandler
+        (
+            new List<HttpResponseMessage>
+            {
+                NotModified,
+                Ok
+            }
+        );
+        var httpClient = new HttpClient(messageHandler);
+        httpClient.BaseAddress = new Uri("http://localhost:8080/");
+
+        var apiClient = new UnleashApiClient(
+            httpClient,
+            A.Fake<IJsonSerializer>(),
+            A.Fake<UnleashApiClientRequestHeaders>(),
+            new EventCallbackConfig()
+        );
+
+        // Act
+        var result = Task.Run(() => apiClient.FetchToggles("etag", CancellationToken.None))
+            .GetAwaiter()
+            .GetResult();
+        var result2 = Task.Run(() => apiClient.FetchToggles("etag", CancellationToken.None))
+            .GetAwaiter()
+            .GetResult();
+        
+        // Assert
+        messageHandler.CallCount.Should().Be(2);
+    }
+
+    [Test]
+    public void Should_Skip_One_After_First_429()
+    {
+        // Arrange
+        var messageHandler = new CountingConfigurableHttpMessageHandler
+        (
+            new List<HttpResponseMessage>
+            {
+                TooManyRequests,
+                Ok
+            }
+        );
+        var apiClient = GetClient(messageHandler);
+
+        // Act 1
+        var result = Task.Run(() => apiClient.FetchToggles("etag", CancellationToken.None))
+            .GetAwaiter()
+            .GetResult();
+
+        // Assert 1
+        messageHandler.CallCount.Should().Be(1);
+
+        // Act 2
+        var result2 = Task.Run(() => apiClient.FetchToggles("etag", CancellationToken.None))
+            .GetAwaiter()
+            .GetResult();
+
+        // Assert 2
+        messageHandler.CallCount.Should().Be(1);
+
+        // Act 3
+        var result3 = Task.Run(() => apiClient.FetchToggles("etag", CancellationToken.None))
+            .GetAwaiter()
+            .GetResult();
+
+        // Assert 3
+        messageHandler.CallCount.Should().Be(2);
+    }
+
+    [Test]
+    public void Backoff_Caps_Out_At_10() {
+        // Arrange
+        var messageHandler = new CountingConfigurableHttpMessageHandler
+        (
+            new List<HttpResponseMessage>
+            {
+                TooManyRequests,
```

Comment:
Wouldn't we need one more TooManyRequests for this to check that it caps at 10? I count 10 in that list and then Ok responses

Reply:
Good catch, I think I had that at some point? I'll fix! thanks!

Reply:
Ok I've added an 11th bad response and slightly improved comments to show what's going on